noobaa_sa/bucket.py

```diff
@@ -87,22 +90,65 @@ def delete(self, bucket_name, config_root=None):
         if config_root is None:
             config_root = self.config_root
         log.info(f"Deleting {bucket_name} Bucket from NSFS")
-        cmd = f"sudo /usr/local/noobaa-core/bin/node {self.manage_nsfs} bucket delete --name {bucket_name} --config_root {config_root}"
+        cmd = f"{self.base_cmd} bucket delete --name {bucket_name} --config_root {config_root}"
         retcode, stdout, stderr = self.conn.exec_cmd(cmd)
         if retcode != 0:
             raise e.BucketDeletionFailed(f"Deleting bucket failed with error {stderr}")
-        
         log.info(stdout)
         log.info("Bucket deleted successfully")
 
 
-    def update(self, bucket_name):
+    def update(self, bucket_name, config_root=None, **kwargs):
         """
         Bucket update
 
         Args:
             bucket_name (str): Bucket name to be updated
+            config_root (str): Path to config root
+        """
+        if config_root is None:
+            config_root = self.config_root
+        self.update_data = kwargs
+        update_cmd = ""         
+        if "new_name" in self.update_data:
+            update_cmd = f"--new_name {self.update_data.get('new_name')}"
+        #if "email" in self.update_data:
+        #    update_cmd = f"--email {self.update_data.get('email')}"
+        if "path" in self.update_data:
+            update_cmd = f"--path {self.update_data.get('path')}"
+        log.info(f"Updating the bucket \"{bucket_name}\"")
+        cmd = f"{self.base_cmd} bucket update --name {bucket_name} {update_cmd} --config_root {config_root} {self.unwanted_log}"
+        retcode, stdout, stderr = self.conn.exec_cmd(cmd)
+        if retcode != 0:
+            raise e.BucketUpdateFailed(f"Failed to update bucket \"{bucket_name}\" with error {stderr}")
+        log.info(stdout)
+        log.info("Bucket info updated successfully")
+        # TODO: Implement --bucket_policy update operation
+        # TODO: Implement --fs_backend update operation 
+        if "new_name" in self.update_data:
+            """
+            Update bucket name with original name
+            """
+            log.info(f"Changing bucket name back to original name")
+            cmd = f"{self.base_cmd} bucket update --name {self.update_data.get('new_name')} --new_name {bucket_name} --config_root {config_root} {self.unwanted_log}"
+            retcode, _, stderr = self.conn.exec_cmd(cmd)
+            if retcode != 0:
+                raise e.BucketUpdateFailed(f"Failed to update bucket back to original name with error {stderr}")
```

Comment:
In case we would like to have the option to change a bucket's name and keep it that way in the future, I suggest we move this logic that restores the bucket's original name to the test itself.

Reply:
Done